packages/pages/src/common/src/function/internal/types.ts

```diff
@@ -0,0 +1,116 @@
+import { ServerlessFunction, FunctionModule } from "../types.js";
+import {
+  validateFunctionDefaultExport,
+  validateFunctionModule,
+} from "./validateFunctionModule.js";
+import { convertToPluginEvent, PluginEvent } from "../../ci/ci.js";
+
+/**
+ * A domain representation of a serverless function module. Contains all fields from an imported
+ * module as well as metadata about the module used in downstream processing.
+ */
+export interface FunctionModuleInternal {
+  /** The filepath to the serverless function file. */
+  filePath: FunctionFilePath;
+  /** The exported config function */
+  config: FunctionConfigInternal;
+  /** The exported getPath function */
+  getPath?: () => string;
+  /** The exported function */
+  default?: ServerlessFunction;
+  /** The slug to host the function at on the Express dev server */
+  slug: {
+    production: string;
+    dev: string;
+    original: string;
+  };
+}
+
+/**
+ * The exported `config` function's definition.
+ */
+export interface FunctionConfigInternal {
+  /** The given name of the serverless function. */
+  name: string;
+  /** The http event. */
+  event: PluginEvent;
+  /** The default export's name */
+  functionName: string;
+}
+
+/**
+ * Stores the filepath to a function
+ */
+export interface FunctionFilePath {
+  /** The absolute path from the user's root directory */
+  absolute: string;
+  /** The path from src/functions */
+  relative: string;
+  /** The file extensions */
+  extension: string;
+}
+
+/**
+ * Converts user-provided function information to an internal module
+ * @param functionFilepath the filepath information for the function
+ * @param functionModule the public function module to convert
+ */
+export const convertFunctionModuleToFunctionModuleInternal = (
+  functionFilepath: FunctionFilePath,
+  functionModule: FunctionModule
+): FunctionModuleInternal => {
+  let functionInternal;
+
+  const functionType = functionFilepath.relative.split("/")[1];
+
+  if (
+    (functionType === "http" ||
+      functionType === "onUrlChange" ||
+      functionType === "onPageGenerate") &&
+    Object.keys(functionModule).length === 1
+  ) {
+    validateFunctionDefaultExport(functionFilepath.absolute, functionModule);
+
+    const defaultSlug = functionFilepath.relative.replace(
```

Comment:
To my previous question, it looks like getPath does not override here. We should also ask Product if we want this at all. I could see value in it, but it might be better to start with less configurability to keep it simpler for users and documentation.